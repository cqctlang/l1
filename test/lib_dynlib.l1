
@include <dynlib.cqct>

lp = loadpath();
str = lp[0] + "/../test/libs";
append(lp,str);
setloadpath(lp);

stderr = open("/dev/null","w");

@dynlib( main, 
  [ (f1, lib1.f2), //this function exists and lib1 depends on lib2
    (f2, lib_.f2), //library does not exist!
    (f3, lib1.f2_), //function does not exist!
  ], [] ) 
{
  printf("In main\n"); 
  f1();
  /* these include a full path in the regression path.
  applyk( f2, 
      @lambda(ret...) { printf("This shouldn't happen!\n"); },
      @lambda() { printf("Successfully caused an error\n"); },
      []);
  applyk( f3, 
      @lambda(ret...) { printf("This shouldn't happen!\n"); },
      @lambda() { printf("Successfully caused an error\n"); },
      []);
  */
}
