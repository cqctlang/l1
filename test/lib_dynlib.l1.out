In main
in lib1
Running f1 from lib1
In lib2: Running f1 from lib1
Finished lib2
Running f1 from lib2
Running f4 from lib2
Running f2 from lib1
finished with lib1
Running f1 from lib1
Running f1 from lib2
Running f4 from lib2
Running f2 from lib1
